lib/connextclient/ConnextClient.ts

```diff
@@ -0,0 +1,491 @@
+  }
+
+  /**
+   * Checks for connectivity and gets our Connext account address
+   */
+  public initSpecific = async () => {
+    if (!this.wallet) {
+      throw errors.CONNEXT_WALLET_NOT_INITIATED;
+    }
+    if (this.wallet) {
+      await this.initConnextClient();
+      this.setStatus(ClientStatus.Initialized);
+      this.emit('initialized');
+      await this.verifyConnectionWithTimeout();
+    }
+  }
+  /**
+   * Initiates wallet for the Connext client
+   */
+  public initWallet = async (seedMnemonic: string[]) => {
+    this.wallet = new ConnextWallet(seedMnemonic.join(' '));
+  }
+
+  public initConnextClient = async () => {
+    if (!this.wallet) {
+      throw errors.CONNEXT_WALLET_NOT_INITIATED;
+    }
+    const wallet = this.wallet;
+    const client = await connext.connect({
+      ethProviderUrl: this.ethProviderUrl,
+      nodeUrl: this.nodeUrl,
+      xpub: wallet.xpub,
+      keyGen: (index: string) => wallet.keyGen(index),
+      store: new ConnextStore(new FileStorage()),
+    });
+    this.client = client;
+    return client;
+  }
+  /**
+   * Associate connext with currencies that have a token address
+   */
+  private setTokenAddresses = (currencyInstances: CurrencyInstance[]) => {
+    currencyInstances.forEach((currency) => {
+      if (currency.tokenAddress) {
+        this.tokenAddresses.set(currency.id, currency.tokenAddress);
+      }
+    });
+  }
+
+  public totalOutboundAmount = (): number => {
+    // assume MAX_AMOUNT since Connext will re-collaterize accordingly
+    return MAX_AMOUNT;
+  }
+
+  public maxChannelOutboundAmount = (): number => {
+    // assume MAX_AMOUNT since Connext will re-collaterize accordingly
+    return MAX_AMOUNT;
+  }
+
+  public maxChannelInboundAmount = (): number => {
+    // assume MAX_AMOUNT since Connext will re-collaterize accordingly
+    return MAX_AMOUNT;
+  }
+
+  protected updateCapacity = async () => {
+    try {
+      const channelBalancePromises = [];
+      for (const [currency] of this.tokenAddresses) {
+        channelBalancePromises.push(this.channelBalance(currency));
+      }
+      await Promise.all(channelBalancePromises);
+    } catch (e) {
+      this.logger.error('failed to update total outbound capacity', e);
+    }
+  }
+
+  protected getConnextClient(): IConnextClient {
+    if (!this.client) {
+      throw errors.CONNEXT_CLIENT_NOT_INITIATED;
+    }
+    return this.client;
+  }
+
+  protected getTokenAddress(currency: string): string {
+    const tokenAdress = this.tokenAddresses.get(currency);
+    if (!tokenAdress) {
+      throw errors.TOKEN_ADDRESS_NOT_FOUND;
+    }
+    return tokenAdress;
+  }
+
+  protected verifyConnection = async () => {
+    this.logger.info('trying to verify connection to connext');
+    try {
+      const client = this.getConnextClient();
+
+      await client.isAvailable();
+
+      this.emit('connectionVerified', { newIdentifier: client.publicIdentifier });
+      this.setStatus(ClientStatus.ConnectionVerified);
```

Comment:
This is a change I made recently, but I made a `setConnected` method that handles emitting the connected event and updating the status at once - in the future if there are any additional steps involved with updating status to connected we can just update that method. Can you use that here instead of these two lines?